src/Compressors/count_sketch.jl

```diff
@@ -0,0 +1,176 @@
+    # Check on the compression dimension
+    function CountSketch(cardinality, compression_dim, type)
+        if compression_dim <= 0
+            throw(ArgumentError("Field 'compression_dim' must be positive."))
+        end
+
+        return new(cardinality, compression_dim, type)
+    end
+end
+
+function CountSketch(;
+    cardinality::Cardinality=Left(),
+    compression_dim::Int64=2,
+    type::Type{<:Number}=Float64
+)
+    # Partially construct the count sketch datatype
+    return CountSketch(cardinality, compression_dim, type)
+end
+
+"""
+    CountSketchRecipe <: CountSketchRecipe
+
+The recipe containing all allocations and information for the CountSketch compressor.
+
+# Fields
+- `cardinality::C where C<:Cardinality`, the cardinality of the compressor. The
+value is either `Left()` or `Right()`.
+- `compression_dim::Int64`, the target compression dimension.
+- `n_rows::Int64`, the number of rows of the compression matrix.
+- `n_cols::Int64``, the number of columns of the compression matrix.
+- `mat::SparseMatrixCSC`, the compression matrix stored in a sparse form.
+"""
+mutable struct CountSketchRecipe{C<:Cardinality} <: CompressorRecipe
+    cardinality::Cardinality
+    compression_dim::Int64
+    n_rows::Int64
+    n_cols::Int64
+    mat::SparseMatrixCSC
+end
+
+function CountSketchRecipe(
+    cardinality::Left,
+    compression_dim::Int64,
+    A::AbstractMatrix,
+    type::Type{<:Number},
+)
+    # determine the initial size
+    n_rows = compression_dim
+    n_cols = size(A, 1)
+    initial_size = n_cols
+    # assign -1 or +1 in every row/column with probability 0.5
+    signs = convert(Vector{type}, rand([-1.0, 1.0], initial_size))
+    groups = rand(1:compression_dim, initial_size)
+    ptr = collect(1: initial_size)
+    mat = sparse(groups, ptr, signs, n_rows, n_cols)
+    return CountSketchRecipe{Left}(cardinality, compression_dim, n_rows, n_cols, mat)
+end
+
+function CountSketchRecipe(
+    cardinality::Right,
+    compression_dim::Int64,
+    A::AbstractMatrix,
+    type::Type{<:Number},
+)
+    # determine the initial size
+    n_rows = size(A, 2)
+    n_cols = compression_dim
+    initial_size = n_rows
+    # assign -1 or +1 in every row/column with probability 0.5
+    signs = convert(Vector{type}, rand([-1.0, 1.0], initial_size))
+    groups = rand(1:compression_dim, initial_size)
+    ptr = collect(1: initial_size)
+    mat = sparse(groups, ptr, signs, n_cols, n_rows)'
+    return CountSketchRecipe{Right}(cardinality, compression_dim, n_rows, n_cols, mat)
+end
+
+function complete_compressor(ingredients::CountSketch, A::AbstractMatrix)
+    return CountSketchRecipe(
+        ingredients.cardinality,
+        ingredients.compression_dim,
+        A,
+        ingredients.type,
+    )
+end
+
+function update_compressor!(S::CountSketchRecipe)
+    # Assign -1 or +1 in every row/column with probability 0.5
+    rand!(S.mat.nzval, [-1.0, 1.0])
+    rand!(S.mat.rowval, 1:S.compression_dim)
+    return nothing
+end
```

Comment:
We will need to check that when the Count Sketch is from `Right()` that this updates correctly as we are updating the columns not the rows of the matrix.